atlasdb-commons/src/main/java/com/palantir/util/result/Result.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.util.result;
+
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+
+public abstract class Result<T, E> {
+
+    private Result() {}
+
+    public static final class Ok<T, E> extends Result<T, E> {
+        private final T value;
+
+        public Ok(T value) {
+            this.value = value;
+        }
+
+        public T getValue() {
+            return value;
+        }
+    }
+
+    public static final class Err<T, E> extends Result<T, E> {
```

Comment:
nit: I'd als opt for `Error` over `Err`, although I recognize other languages probably to the latter. But from a style standpoint,  I think we mostly bias to full words rather than shortened.